### PR TITLE
PAPP-6951: Symantec DLP: Added 2 new actions

### PR DESCRIPTION
**Pull request by jgagan-crest**
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [x] Testing of all the changes has been performed (for bug fixes / features)
- [x] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [x] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [x] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [x] Verify all checks are passing.
- [x] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [x] Do NOT submit updates to dependencies unless it fixes an issue.
- [x] Limit your pull request to one application and submit multiple pull requests if necessary.
#### Please check the type of change your PR introduces:
- [ ] New App
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe): 

## Release Notes (REQUIRED)
- Added 'get incident' and 'list incidents' actions [[PAPP-7011](https://splunk.atlassian.net/browse/PAPP-7011)]

### CODE REVIEW

File: readme.html [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <!-- File: readme.html
-  Copyright (c) 2018-2021 Splunk Inc.
+  Copyright (c) 2018-2022 Splunk Inc.
 
 Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
```
Comment (ngaglani-crest) [inline readme.html:5]: Add HTTP/ HTTPS port details in this file.

File: symantecdlp.json [hunk 2]
```diff
@@ -297,6 +297,679 @@
...
+                        1
+                    ]
+                }
+            ],
+            "versions": "EQ(*)"
+        },
+        {
+            "action": "get incident",
+            "description": "Get DLP incident",
+            "verbose": "User has to provide value for either only one from <b>incident_id</b> or <b>incident_long_id</b> to get the incident details. It is recommended to use <b>incident_long_id</b> parameter to fetch incident details.",
```
Comment (ngaglani-crest) [inline symantecdlp.json:392]:
```suggestion
            "verbose": "User needs to provide value for either <b>incident_id</b> or <b>incident_long_id</b> to get the incident details. It is recommended to use <b>incident_long_id</b> parameter to fetch incident details.",
```

File: symantecdlp_connector.py [hunk 3]
```diff
@@ -338,11 +346,126 @@ def _handle_update_incident(self, param):
...
+                'Please provide only one of the value for incident_long_id or incident_id not both.'
+            )
+
+        include_violations = param.get(DLP_JSON_INCLUDE_VIOLATIONS)
+        include_history = param.get(DLP_JSON_INCLUDE_HISTORY)
+        ret_val, incident_detail = self._get_incident_details(
+            action_result, incident_id, incident_long_id, config[DLP_JSON_EXTRACT_COMPONENTS], include_violations, include_history
+        )
+
+        if (phantom.is_fail(ret_val)):
```
Comment (ngaglani-crest) [inline symantecdlp_connector.py:457]: Please remove the redundant parentheses.

Comment (jgagan-crest) [reply symantecdlp_connector.py]: done

File: symantecdlp.json (hunk 2 above)
Comment (jgagan-crest) [reply symantecdlp.json]: done

File: symantecdlp_connector.py [hunk 4]
```diff
@@ -338,11 +346,126 @@ def _handle_update_incident(self, param):
...
+        # create zeep client
+        ret_val = self._create_client(action_result)
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        config = self.get_config()
+        if (param.get(DLP_JSON_INCIDENT_LONG_ID) is None) and (param.get(DLP_JSON_INCIDENT_ID) is None):
+            return action_result.set_status(
+                phantom.APP_ERROR,
+                'Please provide value for atleast one of the parameter incident_long_id or incident_id.'
```
Comment (rgil-splunk) [inline symantecdlp_connector.py:432]:
```suggestion
                'Please provide a value for at least one of the parameters: incident_long_id or incident_id.'
```

File: symantecdlp_connector.py [hunk 5]
```diff
@@ -338,11 +346,126 @@ def _handle_update_incident(self, param):
...
+
+        incident_id = self._validate_integers(action_result, param.get(DLP_JSON_INCIDENT_ID), DLP_JSON_INCIDENT_ID)
+
+        if not (param.get(DLP_JSON_INCIDENT_ID) is None) and (incident_id is None):
+            return action_result.get_status()
+
+        if not (incident_long_id is None) and (not (incident_id is None)):
+            return action_result.set_status(
+                phantom.APP_ERROR,
+                'Please provide only one of the value for incident_long_id or incident_id not both.'
```
Comment (rgil-splunk) [inline symantecdlp_connector.py:448]:
```suggestion
                'Please specify only one of the following parameters: incident_long_id or incident_id.'
```

File: symantecdlp_connector.py [hunk 6]
```diff
@@ -338,11 +346,126 @@ def _handle_update_incident(self, param):
...
+        config = self.get_config()
+        if (param.get(DLP_JSON_INCIDENT_LONG_ID) is None) and (param.get(DLP_JSON_INCIDENT_ID) is None):
+            return action_result.set_status(
+                phantom.APP_ERROR,
+                'Please provide value for atleast one of the parameter incident_long_id or incident_id.'
+            )
+
+        incident_long_id = self._validate_integers(action_result, param.get(DLP_JSON_INCIDENT_LONG_ID), DLP_JSON_INCIDENT_LONG_ID)
+
+        if not (param.get(DLP_JSON_INCIDENT_LONG_ID) is None) and (incident_long_id is None):
```
Comment (rgil-splunk) [inline symantecdlp_connector.py:437]:
```suggestion
        if param.get(DLP_JSON_INCIDENT_LONG_ID) is not None and incident_long_id is None:
```

File: symantecdlp_connector.py [hunk 7]
```diff
@@ -338,11 +346,126 @@ def _handle_update_incident(self, param):
...
+            return action_result.get_status()
+
+        incident_id = self._validate_integers(action_result, param.get(DLP_JSON_INCIDENT_ID), DLP_JSON_INCIDENT_ID)
+
+        if not (param.get(DLP_JSON_INCIDENT_ID) is None) and (incident_id is None):
```
Comment (rgil-splunk) [inline symantecdlp_connector.py:442]:
```suggestion
        if param.get(DLP_JSON_INCIDENT_ID) is not None and incident_id is None:
```

File: symantecdlp_connector.py [hunk 8]
```diff
@@ -338,11 +346,126 @@ def _handle_update_incident(self, param):
...
+            return action_result.get_status()
+
+        if not (incident_long_id is None) and (not (incident_id is None)):
```
Comment (rgil-splunk) [inline symantecdlp_connector.py:445]:
```suggestion
        if incident_long_id is not None and incident_id is not None:
```

File: symantecdlp_connector.py [hunk 9]
```diff
@@ -338,11 +346,126 @@ def _handle_update_incident(self, param):
...
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        # create zeep client
+        ret_val = self._create_client(action_result)
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        config = self.get_config()
+        if (param.get(DLP_JSON_INCIDENT_LONG_ID) is None) and (param.get(DLP_JSON_INCIDENT_ID) is None):
```
Comment (rgil-splunk) [inline symantecdlp_connector.py:429]:
```suggestion
        if param.get(DLP_JSON_INCIDENT_LONG_ID) is None and param.get(DLP_JSON_INCIDENT_ID) is None:
```

File: symantecdlp_connector.py [hunk 10]
```diff
@@ -373,10 +496,17 @@ def _zeep_to_dict(self, resp):
         response = self._sanitize_dict(response)
         return response
 
-    def _get_incident_details(self, action_result, incident_id, extract_files):
+    def _get_incident_details(self, action_result, incident_id, incident_long_id, extract_files, includeViolations, includeHistory):
 
+        binary_incident_id = incident_id
+        kwargs = dict()
+        if not(incident_long_id is None):
```
Comment (rgil-splunk) [inline symantecdlp_connector.py:503]:
```suggestion
        if incident_long_id is not None:
```

**Review by rgil-splunk — APPROVED** [review]

File: symantecdlp.json [hunk 11]
```diff
@@ -297,6 +297,679 @@
...
+                        "dlp incident id"
+                    ]
+                },
+                "include_violations": {
+                    "description": "Whether the Web Service should return policy violation data with the basic incident details",
+                    "data_type": "boolean",
+                    "order": 2
+                },
+                "include_history": {
+                    "description": "Whether the Web Service should returnincident history information",
```
[the remark this answers is not in the document]
Comment (jgagan-crest) [reply symantecdlp.json]: Done

File: symantecdlp.json [hunk 12]
```diff
@@ -297,6 +297,679 @@
...
+                    "data_type": "numeric",
+                    "example_values": [
+                        2701
+                    ]
+                },
+                {
+                    "data_path": "action_result.data.*.incident.recipient.*.recipientIdentifier",
+                    "data_type": "string",
+                    "example_values": [
+                        "http://edgedl.me.gvt1.com/edgedl/delta-update/hfnkpimlhhgieaddgfemjhofmfblmnib/1.16e647889a74b001c56966e6fb7bb6d8d5c5e8272d0df4d46e35034f3b77e001/1.3d372163ae377d432eda14baa4e8f2e58086603347ed3c6a275e0f81030bc7ee/038e31d1bba6578f68cdf133b371b85a01119f2701f599cdfac773ae974cd140.crxd"
```
[the remark this answers is not in the document]
Comment (jgagan-crest) [reply symantecdlp.json]: Done

File: symantecdlp_connector.py [hunk 13]
```diff
@@ -338,11 +346,126 @@ def _handle_update_incident(self, param):
...
+
+        date_status = self._validate_date(creation_time_after)
+        if not date_status:
+            return action_result.set_status(phantom.APP_ERROR, DLP_TIMESTAMP_VALIDATION_FAILED_MSG)
+
+        try:
+            response = self._client.service.incidentList(report_id, creation_time_after)
+        except Exception as e:
+            message = 'SOAP call to DLP failed. {}'.format(self._get_error_message_from_exception(e))
+            return RetVal(action_result.set_status(phantom.APP_ERROR, "Error Occored: {}".format(message)))
```
[the remark this answers is not in the document]
Comment (jgagan-crest) [reply symantecdlp_connector.py]: Done

File: symantecdlp_connector.py (hunk 4 above)
Comment (jgagan-crest) [reply symantecdlp_connector.py]: Done

File: symantecdlp_connector.py (hunk 5 above)
Comment (jgagan-crest) [reply symantecdlp_connector.py]: Done

File: symantecdlp_connector.py (hunk 6 above)
Comment (jgagan-crest) [reply symantecdlp_connector.py]: Done

File: symantecdlp_connector.py (hunk 8 above)
Comment (jgagan-crest) [reply symantecdlp_connector.py]: Done

File: symantecdlp_connector.py (hunk 9 above)
Comment (jgagan-crest) [reply symantecdlp_connector.py]: Done

File: symantecdlp_connector.py (hunk 10 above)
Comment (jgagan-crest) [reply symantecdlp_connector.py]: Done

File: symantecdlp_connector.py (hunk 7 above)
Comment (jgagan-crest) [reply symantecdlp_connector.py]: Done

**Review by ngaglani-crest — APPROVED** [review]

**Pull request merged**